### DISCUSSION TITLE
DSA or Development ?

### DISCUSSION THREAD

**Discussion by SyedImtiyaz-1** [Schweinepriester/github-profile-achievements#782] [Q&A]
Hello Everyone,
DSA or Development which is more important for placements ?
Suggest me :)

**Comment by MrAlaminH** [comment]
Both DSA (Data Structures and Algorithms) and Development are important for placements, but their importance can vary based on the role and company.

DSA is crucial for understanding how to efficiently store, access, and manipulate data, which is a fundamental aspect of programming. It is especially important for roles focused on backend development, systems, or algorithms. Many tech companies, especially large ones like Google, Amazon, and Microsoft, emphasize DSA in their interview process.

On the other hand, Development skills are important for building applications and systems. This includes frontend and backend web development, mobile app development, etc. Companies that are looking for role-specific skills, like a Frontend Developer or a Backend Developer, might focus more on your development skills.

In general, having a good balance of both DSA and Development skills would make you a well-rounded candidate. However, you should focus more on one or the other based on the specific roles and companies you are interested in.

**Comment by aftab-ahmad-khan-dev** [comment]
### Short answer: do both, but sequence them

If your goal is a job / freelancing income soon:
1. **Development first (8–12 weeks)** — ship 2–3 real projects (CRUD + auth + deploy). Portfolio beats theory in hiring for most junior/mid roles.
2. **DSA in parallel lightly** — 3–5 problems/week (arrays, hash maps, two pointers). Enough for interviews without stalling shipping.

If your goal is FAANG-style interviews first:
- Invert the ratio: heavier DSA + system design, but still keep one live project so you can talk about production tradeoffs.

Practical weekly split many people sustain:
- 60% building (features, PRs, debugging)
- 30% DSA
- 10% reading docs / system design

For MERN / Shopify freelancing specifically: **projects + client communication** matter more than LeetCode volume.
